### PR TITLE
 Free up disk space post building site by removing docs-archive directory

### PR DESCRIPTION
**Pull request by pankajkoti**
I did a search in the repo and found there is no other reference to the `docs-archive` directory after the site is built, so I believe we're safe to remove the directory contents from the CI job for the subsequent steps once the site is built. This step kept before the `Deploy website on asf-site branch` step will ensure (due to 10GB+ space reclaimed) that CI build will not fail as observed last time in the [Github action run](https://github.com/apache/airflow-site/actions/runs/4243773508/jobs/7376961974) due to the `No space left on device` error. 

related to: #773

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -52,21 +62,37 @@ jobs:
...
       - name: 🔎 Lint
         uses: apache/airflow-pre-commit-action@0764670bf370aab253130d534e1eda7ff497dc60  # v2.0.0
       - name: 🔧 Build site
         run: ./site.sh build-site
+      - run: |
+          echo "Disk free (df -h)"
+          df -h
+      - run: |
+          rm -rf ./docs-archive/*
+          echo "These files have been deleted as they are no longer needed after the site is built in the above step"
```
Comment (jedcunningham) [inline .github/workflows/build.yml:84]: This part LGTM! Let's PR this!

**Comment by potiuk** [comment]
Why don't you leave the`df` commands? They are useful. It would be maybe helpfuil to wrap them in a script and add grouping so that they are folded by default to not spoil the output ?: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#grouping-log-lines

**Comment by jedcunningham** [comment]
It feels a little noisy to me. I'm not sure we need it between every step, but if we want to add it in a select few places, no real objections.

File: .github/scripts/group_bash_logs.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+# Create an expandable group in the logs for the bash command output using the syntax defined for GiHub Action runners:
+# https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#grouping-log-lines.
+# With such expandable logs groupings we can avoid noisy outputs and add the same time have pointers for
+# debugging/analysis.
+function run_and_group_bash_logs {
```
Comment (potiuk) [inline .github/scripts/group_bash_logs.sh:25]: I'd just make "print_df.sh" script :). No need to generalize it. It would be quite a bit simpler.

**Comment by potiuk** [comment]
BTW. Sorry for the extra work my request caused, but I hope that might be a good learning for the future GitHub Actions changes - we have lots and lots of those in the main repo :)

**Comment by pankajkoti** [comment]
> BTW. Sorry for the extra work my request caused, but I hope that might be a good learning for the future GitHub Actions changes - we have lots and lots of those in the main repo :)

Yes Jarek, now it's green, appreciate amazing quick reviews, and sorry the failures spammed you on the CI. I am hoping to learn more in the coming year targeting more Airflow OSS contributions and reaching 1 level up if not more by learning from the PR review suggestions :)

File: .github/scripts/group_bash_logs.sh (hunk 2 above)
Comment (pankajkoti) [reply .github/scripts/group_bash_logs.sh]: addressed, please check if that's what you meant to suggest

**Comment by pankajkoti** [comment]
It's ready for review now. Please check when you have time

@potiuk @jedcunningham

**Comment by potiuk** [comment]
Nice. I love it!

**Review by potiuk — APPROVED** [review]

**Review by jedcunningham — APPROVED** [review]

**Comment by jedcunningham** [comment]
Thanks @pankajkoti 🎉

**Comment by pankajkoti** [comment]
Thank you @potiuk @jedcunningham @phanikumv for giving me an opportunity to work on this iteration of the story, has been a nice learning, and it feels great! Looking forward to contributing ahead and continuously!

**Pull request merged**